File: PharmaDicBackEnd/PharmaDicBackEnd.ApiService/Models/SearchHistory.cs
using System;
using System.Collections.Generic;

namespace PharmaDicBackEnd.ApiService.Models;

public partial class SearchHistory
{
    public int SearchId { get; set; }

    public int? UserId { get; set; }

    public string? Keyword { get; set; }

    public string? SearchType { get; set; }

    public DateTime? SearchDate { get; set; }

    public virtual User? User { get; set; }
}

File: PharmaDicMobile/app/src/main/java/com/example/pharmadicmobile/models/SearchHistory.java
package com.example.pharmadicmobile.models;

import java.io.Serializable;
import java.util.Date;

public class SearchHistory implements Serializable {
    private int searchId;
    private Integer userId;
    private String keyword;
    private String searchType;
    private Date searchDate;
    private User user;

    public SearchHistory() {}

    public int getSearchId() { return searchId; }
    public void setSearchId(int searchId) { this.searchId = searchId; }

    public Integer getUserId() { return userId; }
    public void setUserId(Integer userId) { this.userId = userId; }

    public String getKeyword() { return keyword; }
    public void setKeyword(String keyword) { this.keyword = keyword; }

    public String getSearchType() { return searchType; }
    public void setSearchType(String searchType) { this.searchType = searchType; }

    public Date getSearchDate() { return searchDate; }
    public void setSearchDate(Date searchDate) { this.searchDate = searchDate; }

    public User getUser() { return user; }
    public void setUser(User user) { this.user = user; }
}
